web/js/codeworld.js

```diff
@@ -158,6 +158,63 @@ function initCodeworld() {
     }
 }
 
+var cStream;
+var recorder;
+var chunks = [];
+
+function captureStart() {
+    var iframe = document.querySelector('#runner');
+    var innerDoc = iframe.contentDocument || iframe.contentWindow.document;
+
+    var canvas = innerDoc.querySelector('#screen');
+
+    // Recording Framerate
+    var cStream = canvas.captureStream(30);
+
+    recorder = new MediaRecorder(cStream);
+    recorder.start();
+
+    document.querySelector('#recordIcon').style.display = '';
+    document.querySelector('#startRecButton').style.display = 'none';
+    document.querySelector('#stopRecButton').style.display = '';
+
+    recorder.ondataavailable = (function(e) { chunks.push(e.data); });
+
+    recorder.onstop = (function() { exportStream(chunks); });
+};
+
+function stopRecording() {
+    recorder.stop();
+    document.querySelector('#recordIcon').style.display = 'none';
+    document.querySelector('#startRecButton').style.display = '';
+    document.querySelector('#stopRecButton').style.display = 'none';
+}
+
+function exportStream(e) {
+    var blob = new Blob(chunks);
+    chunks = [];
+
+    var d = new Date();
+    var videoFileName = 'codeworld_recording_'
+                        + d.toDateString().split(' ').join('_') + '_'
+                        + d.getHours() +':'+ d.getMinutes() +':'+ d.getSeconds()
+                        +'.webm';
```

Comment:
You haven't specified the format in options to MediaRecorder.  Are you sure the stream is in webm format?  I don't know this API too well.